packages/jest-types/src/Config.ts

```diff
@@ -254,6 +254,69 @@ type ShardConfig = {
   shardCount: number;
 };
 
+export type PluginContext = {
+  config: ProjectConfig;
+  globalConfig: GlobalConfig;
+};
+
+export type ResolveIdResult =
+  | {
+      id: string;
+      external?: boolean;
+    }
+  | string
+  | null
+  | undefined;
+
+export type LoadResult =
+  | {
+      code: string;
+      map?: any;
+    }
+  | string
+  | null
+  | undefined;
+
+export type TransformPluginResult =
+  | {
+      code: string;
+      map?: any;
+    }
+  | string
+  | null
+  | undefined;
+
+export interface JestPlugin {
+  name: string;
+  config?: (
+    config: InitialOptions,
+    context: {configPath: string | null},
+  ) => InitialOptions | Promise<InitialOptions> | void | Promise<void>;
+  configResolved?: (
+    config: ProjectConfig,
+    context: PluginContext,
+  ) => void | Promise<void>;
+  configureJest?: (context: PluginContext) => void | Promise<void>;
+  resolveId?: (
+    source: string,
+    importer: string | undefined,
+    options: {isEntry: boolean},
+  ) => ResolveIdResult | Promise<ResolveIdResult>;
+  load?: (id: string) => LoadResult | Promise<LoadResult>;
+  transform?: (
+    code: string,
+    id: string,
+  ) => TransformPluginResult | Promise<TransformPluginResult>;
+}
+
+// Alias for backward compatibility
+export type Plugin = JestPlugin;
```

Comment:
Please remove this, we don't need. If any places are using your new interface, please replace with `JestPlugin` interface